src/theme-context-provider.tsx

```diff
@@ -165,44 +176,51 @@ const ThemeContextProvider: React.FC<Props> = ({theme, children, as, withoutStyl
-        </TabFocus>
+        <>
+            <TabFocus disabled={!theme.enableTabFocus}>
+                <ModalContextProvider>
+                    <TooltipContextProvider>
+                        <ThemeContext.Provider value={contextTheme}>
+                            <TrackingConfig eventFormat={contextTheme.analytics.eventFormat}>
+                                <AspectRatioSupportProvider>
+                                    <DocumentVisibilityProvider>
+                                        <AriaIdGetterContext.Provider value={getAriaId}>
+                                            <ScreenSizeContextProvider>
+                                                <DialogRoot>
+                                                    <SnackbarRoot>
+                                                        {as ? (
+                                                            React.createElement(
+                                                                as,
+                                                                {
+                                                                    style: withoutStyles
+                                                                        ? undefined
+                                                                        : themeVars,
+                                                                },
+                                                                children
+                                                            )
+                                                        ) : (
+                                                            <>
+                                                                {!withoutStyles &&
+                                                                    (process.env.NODE_ENV !== 'test' ||
+                                                                        process.env.SSR_TEST) && (
+                                                                        <style>{`:root {${themeVars}}`}</style>
+                                                                    )}
+                                                                {children}
+                                                            </>
+                                                        )}
+                                                    </SnackbarRoot>
+                                                </DialogRoot>
+                                            </ScreenSizeContextProvider>
+                                        </AriaIdGetterContext.Provider>
+                                    </DocumentVisibilityProvider>
+                                </AspectRatioSupportProvider>
+                            </TrackingConfig>
+                        </ThemeContext.Provider>
+                    </TooltipContextProvider>
+                </ModalContextProvider>
+            </TabFocus>
+            <div ref={ref} style={{display: 'none'}} />
```

Comment:
consider this:
```suggestion
            {process.env.NODE_ENV !== 'test' && <div ref={ref} style={{display: 'none'}} />}
```
This way it won't be inserted in unit tests and won't pollute possible snapshots (like in button-test.tsx)

Reply:
also, this div could be removed from the dom once we have done the job

Reply:
I've removed the div after applying the hack. This fixes also the issue with unit tests, so there is no need for the `NODE_ENV === 'test'` thing